src/layout/commands.rs

```diff
@@ -368,4 +370,26 @@ impl Tree {
         self.0.send_active_to_workspace(workspace_name);
         Ok(())
     }
+
+    /// Resizes the container, as if it was dragged at the edge to a certain point
+    /// on the screen.
+    pub fn resize_container(&mut self, id: Uuid, edge: ResizeEdge, pointer: Point) -> CommandResult {
+        if let Ok(mut lock) = try_lock_action() {
+            if let Some(ref mut action) = *lock {
+                if try!(self.0.lookup(id)).floating() {
+                    self.0.resize_floating(id, edge, pointer, action)
+                } else {
+                    self.0.resize_tiled(id, edge, pointer, action)
+                }
+            } else {
+                // TODO errors
```

Comment:
Ugh, it was annoying, but I integrated action errors in.

I'm thinking I'll need to do a pass where I remove all error handling (and maybe some of the more complex logic) from the commands wrapper, because that's starting to get a bit annoying.

:heavy_check_mark: